administrator/components/com_installer/Controller/Downloadkey.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_installer
+ *
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Installer\Administrator\Controller;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Controller\Form;
+
+/**
+ * Controller for a single contact
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Downloadkey extends Form
+{
+	/**
+	 * Add the prefix and suffix of the download key before saving it into extra_query
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function save()
+	{
+		$prefix = $this->input->getString('dlidprefix');
+		$sufix = $this->input->getString('dlidsufix');
+
+		$input = $this->input->post->get('jform', array(), 'array');
+		$input['extra_query'] = $prefix . $input['extra_query'] . $sufix;
```

Comment:
This is a debug I made in the Model of Downloadkey:
![screenshot from 2017-07-31 19-11-06](https://user-images.githubusercontent.com/20955884/28791331-04e9f228-7624-11e7-9170-d104bc67116f.png)

![screenshot from 2017-07-31 19-07-10](https://user-images.githubusercontent.com/20955884/28791294-e7207d48-7623-11e7-9fc7-6e40c3455ba2.png)

There is no way here I can get the hidden values (prefix and sufix) without having to make a query to find the xml file again with the update_site_id, any idea?